homeworks/sasha.demidenko_SashaDemi/GamesProject/scripts/games.js

```diff
@@ -0,0 +1,228 @@
+    const cardsPlatform = selectElement('platform');
+    const cardsPublisher = selectElement('publisher');
+    const cardsDeveloper = selectElement('developer');
+    const cardsDate = selectElement('release_date');
+    const descMaxLength = 100;
+
+    cardsTitle.textContent = game.title;
+    cardsTitle.style.cssText = 'height: 90px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap';
+
+    cardsImg.src = game.thumbnail;
+    cardsImg.alt = `Thumbnail for ${game.title}`;
+    cardsImg.style.cssText = 'width: 90px; height: 90px; border-radius: 12px';
+
+    cardsDescription.textContent = game.short_description;
+    if (game.short_description.length > descMaxLength) {
+        cardsDescription.dataset.fullText = game.short_description;
+        cardsDescription.textContent = `${game.short_description.substring(0, descMaxLength)}...`;
+        cardsDescription.style.cursor = 'pointer';
+        cardsDescription.addEventListener('click', function () {
+            this.textContent = this.dataset.fullText;
+        });
+    }
+
+    cardsGenre.textContent = `Genre: ${game.genre}`;
+    cardsPlatform.textContent = `Platform: ${game.platform}`;
+    cardsPublisher.textContent = `Publisher: ${game.publisher}`;
+    cardsDeveloper.textContent = `Developer: ${game.developer}`;
+    cardsDate.textContent = `Release Date: ${game.release_date}`;
+
+    return elementCopy;
+}
+
+function renderCards(container, gamesList) {
+    const fragment = document.createDocumentFragment();
+
+    gamesList.forEach((game) => {
+        const card = createCardElement(game);
+        fragment.appendChild(card);
+    });
+
+    container.appendChild(fragment);
+}
+
+function filterGamesByReleaseDate(gamesList, isNewChecked, isOldChecked) {
+    if (isNewChecked && isOldChecked) {
+        return gamesList;
+    }
+
+    return gamesList.filter((game) => {
+        const releaseDate = new Date(game.release_date);
+        const isOld = isOldChecked && releaseDate.getFullYear() >= 2010;
+        const isNew = isNewChecked && releaseDate.getFullYear() <= 2020;
+
+        return !(isNew || isOld);
+    });
+}
+
+function displayCards(container, gamesList, isNewChecked, isOldChecked) {
+    const filteredGames = filterGamesByReleaseDate(gamesList, isNewChecked, isOldChecked);
+
+    const fragment = document.createDocumentFragment();
+
+    filteredGames.forEach((game) => {
+        const card = createCardElement(game);
+        fragment.appendChild(card);
+    });
+
+    container.innerHTML = '';
+    container.appendChild(fragment);
+}
+
+function init() {
+    const newCheckbox = document.getElementById('new');
+    const oldCheckbox = document.getElementById('old');
+    const newCards = document.querySelector('[data-type="cards-container"]');
+    if (newCards) {
+        renderCards(newCards, games);
+    }
+    displayCards(newCards, games, newCheckbox.checked, oldCheckbox.checked);
+
+    newCheckbox.addEventListener('change', init);
```

Comment:
better to use handler function with the name `onFilterCards` or `filterAndRerenderCards`

Reply:
in this handler, you should filter cards and render them